File: src/main/java/com/matchFit/post/dto/response/GetPostsList.java
package com.matchFit.post.dto.response;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class GetPostsList {
    private final List<GetPost> posts;
    private final int page;
    private final int size;
    private final long totalElements;
    private final int totalPages;

    public static GetPostsList from(List<GetPost> posts) {
        return new GetPostsList(posts, 0, posts.size(), posts.size(), 1);
    }

    public static GetPostsList of(
            List<GetPost> posts,
            int page,
            int size,
            long totalElements,
            int totalPages
    ) {
        return new GetPostsList(posts, page, size, totalElements, totalPages);
    }
}

File: src/main/kotlin/com/matchFit/post/dto/response/GetPostsList.kt
package com.matchFit.post.dto.response


data class GetPostsList(
    val posts: List<GetPost>,
    val page: Int = 0,
    val size: Int = posts.size,
    val totalElements: Long = posts.size.toLong(),
    val totalPages: Int = 1
) {
    companion object {
        fun of(posts: List<GetPost>): GetPostsList {
            return GetPostsList(posts)
        }

        fun of(
            posts: List<GetPost>,
            page: Int,
            size: Int,
            totalElements: Long,
            totalPages: Int
        ): GetPostsList {
            return GetPostsList(posts, page, size, totalElements, totalPages)
        }
    }
}
